pkg/operator/controller/ingress/controller.go

```diff
@@ -95,6 +96,15 @@ func New(mgr manager.Manager, config Config) (controller.Controller, error) {
 	if err := c.Watch(&source.Kind{Type: &corev1.Service{}}, enqueueRequestForOwningIngressController(config.Namespace)); err != nil {
 		return nil, err
 	}
+	// Add watch for deleted pods specifically for ensuring ingress deletion.
+	if err := c.Watch(&source.Kind{Type: &corev1.Pod{}}, enqueueRequestForOwningIngressController(config.Namespace), predicate.Funcs{
```

Comment:
watching pods only on delete events so it triggers a reconcilation when they are cleaned up.

Reply:
This should not be needed, as Miciah mentioned.

Reply:
I looked into this again - it does indeed reconcile again on the error "pods aren't deleted", however, it begins to trigger exponential back off, which causes the deletion to be a little slower than it needs to be (5-10 seconds slower).

This watch makes the final reconcile (when all pods are deleted) to be more timely which results in a 5-10 second quicker deletion. I'd say it's worth to keep it in here since that 5-10 seconds can add up.